probe/resolver.go

```diff
@@ -92,13 +92,15 @@ func (r staticResolver) resolveHosts() {
 			}
 		}
 
+		endpoints := make([]string, 0, len(addrs))
 		for _, addr := range addrs {
 			// For now, ignore IPv6
 			if addr.To4() == nil {
 				continue
 			}
-			r.add(net.JoinHostPort(addr.String(), peer.port))
+			endpoints = append(endpoints, net.JoinHostPort(addr.String(), peer.port))
 		}
+		r.set(peer.hostname, endpoints)
```

Comment:
Yes, I'm doing this so that app processes that go away are removed. Yes, if DNS stops returning an IP for a hostname, we will stop pushing reports there until it comes back. But of course we should treat DNS as a source of authority.